codex/stores/repostore/operations.nim

```diff
@@ -0,0 +1,213 @@
+## Nim-Codex
+## Copyright (c) 2024 Status Research & Development GmbH
+## Licensed under either of
+##  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+##  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+## at your option.
+## This file may not be copied, modified, or distributed except according to
+## those terms.
+
+import pkg/chronos
+import pkg/chronos/futures
+import pkg/datastore
+import pkg/datastore/typedds
+import pkg/libp2p/cid
+import pkg/metrics
+import pkg/questionable
+import pkg/questionable/results
+
+import ./coders
+import ./types
+import ../blockstore
+import ../keyutils
+import ../../blocktype
+import ../../clock
+import ../../logutils
+import ../../merkletree
+
+logScope:
+  topics = "codex repostore"
+
+declareGauge(codex_repostore_blocks, "codex repostore blocks")
+declareGauge(codex_repostore_bytes_used, "codex repostore bytes used")
+declareGauge(codex_repostore_bytes_reserved, "codex repostore bytes reserved")
+
+proc putLeafMetadata*(self: RepoStore, treeCid: Cid, index: Natural, blkCid: Cid, proof: CodexProof): Future[?!StoreResultKind] {.async.} =
+  without key =? createBlockCidAndProofMetadataKey(treeCid, index), err:
+    return failure(err)
+
+  await self.metaDs.modifyGet(key,
```

Comment:
If we're simply storing the value, without having to update it, why not just use a simple `put`?

Reply:
Because we need an information if such entry was already in the store or is it a new entry. If it's a new entry we're going to increase block ref count by 1.